lighthouse-core/gather/connections/connection.js

```diff
@@ -100,7 +100,8 @@ class Connection {
       // handleRawError returns or throws synchronously; wrap to put into promise chain.
       return callback.resolve(Promise.resolve().then(_ => {
         if (object.error) {
-          return this.handleRawError(object.error, callback.method);
+          log.formatProtocol('method <= browser ERR', {method: callback.method}, 'error');
+          throw new Error(`Protocol error (${callback.method}): ${object.error.message}`);
```

Comment:
moving this back is basically undoing #861 and #895. `DOM.disable` returns an error if `DOM` is already disabled (see details in #861), but with this change the disable command is never sent if it's already disabled